### PR TITLE
Reconcile dapr service for argo rollout

### PR DESCRIPTION
**Pull request by sunzhaochang**
# Description

Support service reconciler for dapr-enabled argo rollouts.

## Issue reference

Please reference the issue this PR will close: #6034 and #5605 

## Test case
1. Install dapr with `argoRolloutServiceReconciler.enabled` option:
```
helm install --set global.argoRolloutServiceReconciler.enabled=true -f charts/dapr/values.yaml mydapr ./charts/dapr/
```
2. Create dapr-enabled argo rollout resource:
```
cat <<EOF | kubectl apply -f -
apiVersion: argoproj.io/v1alpha1
kind: Rollout
metadata:
  name: rollouts-demo
  namespace: dapr-test
spec:
  replicas: 5
  revisionHistoryLimit: 2
  strategy:
    canary:
      steps:
      - setWeight: 20
      - pause: {duration: 10}
      - setWeight: 80
      - pause: {duration: 10}
  selector:
    matchLabels:
      app: rollouts-demo
  template:
    metadata:
      annotations:
        dapr.io/enabled: "true"
        dapr.io/app-id: "rollouts-demo"
      labels:
        app: rollouts-demo
    spec:
      containers:
      - name: rollouts-demo
        image: argoproj/rollouts-demo:blue
        ports:
        - name: http
          containerPort: 8080
          protocol: TCP
EOF
```
Then the dapr service will be created by operator:
```
$ kubectl get svc -n dapr-test
NAME                 TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)                               AGE
rollouts-demo-dapr   ClusterIP   None         <none>        80/TCP,50001/TCP,50002/TCP,9090/TCP   70s
```
3. Delete the argo rollout resource and dapr service will also be deleted:
```
$ kubectl delete -n dapr-test rollouts.argoproj.io rollouts-demo 
rollout.argoproj.io "rollouts-demo" deleted
$ kubectl get svc -n dapr-test
No resources found in dapr-test namespace.
```
## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: charts/dapr/values.yaml [hunk 1]
```diff
@@ -37,3 +37,5 @@ global:
   #   ca: "ca.crt"
   #   cert: "tls.crt"
   #   key: "tls.key"
+  argoRolloutServiceReconciler:
```
Comment (ItalyPaleAle) [inline charts/dapr/values.yaml:40]: Please update https://github.com/dapr/dapr/blob/master/charts/dapr/README.md too

File: cmd/operator/main.go [hunk 2]
```diff
@@ -119,6 +121,7 @@ func init() {
 	flag.BoolVar(&disableLeaderElection, "disable-leader-election", false, "Disable leader election for operator")
 	flag.BoolVar(&disableServiceReconciler, "disable-service-reconciler", false, "Disable the Service reconciler for Dapr-enabled Deployments and StatefulSets")
 	flag.StringVar(&watchNamespace, "watch-namespace", "", "Namespace to watch Dapr annotated resources in")
+	flag.BoolVar(&enableArgoRolloutServiceReconciler, "enable-argo-rollout-service-reconciler", false, "Enable the service reconciler for dapr-enabled Argo Rollouts")
```
Comment (ItalyPaleAle) [inline cmd/operator/main.go:124]:
```suggestion
	flag.BoolVar(&enableArgoRolloutServiceReconciler, "enable-argo-rollout-service-reconciler", false, "Enable the service reconciler for Dapr-enabled Argo Rollouts")
```

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]
I am not familiar with Argo Rollout so I can only review the code itself, but LGTM with just 2 minor changes please

File: charts/dapr/values.yaml (hunk 1 above)
Comment (sunzhaochang) [reply charts/dapr/values.yaml:40]: Done

**Review by ItalyPaleAle — DISMISSED** [review]

**Review by yaron2 — CHANGES_REQUESTED** [review]

File: pkg/operator/handlers/dapr_handler.go [hunk 3]
```diff
@@ -57,12 +60,13 @@ type Reconciler struct {
 
 // NewDaprHandler returns a new Dapr handler.
 // This is a reconciler that watches all Deployment and StatefulSet resources and ensures that a matching Service resource is deployed to allow Dapr sidecar-to-sidecar communication and access to other ports.
-func NewDaprHandler(mgr ctrl.Manager) *DaprHandler {
+func NewDaprHandler(mgr ctrl.Manager, argoRolloutServiceReconcilerEnabled bool) *DaprHandler {
```
[the remark this answers is not in the document]
Comment (sunzhaochang) [reply pkg/operator/handlers/dapr_handler.go]: I replaced the single param with options struct. Please take a look.

**Review by yaron2 — APPROVED** [review]

**Pull request merged**